main.tf

```diff
@@ -4,7 +4,7 @@ locals {
   iam_role_name              = "castai-eks-${substr(local.resource_name_postfix, 0, 53)}"
   iam_policy_name            = var.create_iam_resources_per_cluster ? "CastEKSPolicy-${local.resource_name_postfix}" : "CastEKSPolicy-tf"
   iam_role_policy_name       = "castai-user-policy-${substr(local.resource_name_postfix, 0, 45)}"
-  instance_profile_role_name = "castai-eks-instance-${substr(local.resource_name_postfix, 0, 44)}"
+  instance_profile_role_name = "cast-${substr(local.resource_name_postfix, 0, 40)}-eks-${substr(var.castai_cluster_id, 0, 8)}"
```

Comment:
instance profile is passed a value to node configuration 
https://github.com/castai/terraform-provider-castai/blob/master/examples/eks/eks_cluster_autoscaler_policies/castai.tf#L80

the change is not necessary

Reply:
a conversation with one of castai member in slack led me to believe that the pattern I suggested was obligatory for the default deployment. so it seems that by using the `castai_node_configuration` resource, defining the name as `default`, you can overwrite the default profile name to a custom one?

Reply:
here you can change the default node configuration https://github.com/castai/terraform-provider-castai/blob/master/examples/eks/eks_cluster_autoscaler_policies/castai.tf#L69 
```
  default_node_configuration_name = "default"
```

`instance_profile_arn` is a field to pass the arn of instance-profile to use for node config